packages/atomic/src/components/atomic-result-list-v1/atomic-result-list.tsx

```diff
@@ -127,7 +127,7 @@ export class AtomicResultList implements InitializableComponent {
   }
 
   private getId(result: Result) {
-    return result.uniqueId + this.resultListState.searchUid;
+    return result.uniqueId + result.searchUid;
```

Comment:
The 1st search returns let's say 10 results, the searchUid is "1"
The 2nd search returns another 10 results, the total is 20, but the searchUid is "2" so all results rerender.

Fix is to had an associated searchUid to the Result object, there might be other ways but it is the cleanest I've found.

If we just use uniqueId, the result won't rerender correctly when doing a query or applying a filter. (e.g. highlighting won't be applied)

Reply:
I want to explore if we can avoid changing the `Result` object. If we created a key that took into account the dynamic parts of a `Result`, wouldn't that solve the problem?

A naive implementation would be to `JSON.stringify` the `Result` object and set that as the key. I think this could perform well since the objects are not that large?

Another approach would be to provide a utility function `getResultId(result: Result)` from headless, that takes 
 a result constructs a key using the dynamic parts.

Reply:
Its not the result that's changing, it's the relation of the result to the query. Doing a naive implementation of the uniqueId + the search params stringified would work, but I did not find that approach clean

Reply:
It also should be saved in the state so a helper on the controller won't be sufficient

Reply:
This makes me realize... 

I think we might have a problem in general with UA for any "show more" kinda request. Normally, we should be reusing the original search ID when loading more page. 

And when we click on a result on page 3, for example, we should log the click event with the search ID that was generated on the first initial query.

Reply:
I don't understand what you mean by the "saved in state"?

In the naive implementation, the key would be `key=JSON.stringify(result)`. If the highlights array changes, the result would change and so would the key.

Alternatively, a more sophisticated implementation would stringify only the dynamic properties of a result instead of the whole object.

Reply:
The previous comment is the more relevant one https://github.com/coveo/ui-kit/pull/1103#discussion_r690607211

Sometimes the result object is the exact same, but we still want the atomic-result component to re-render (e.g. if a field value is selected by a facet, it should appear as selected, etc.) So it's not 100% about the result object itself. It's the relationship to the request/response. For a new query, we should render the results. When we do "load more" the query is the same and the original results should not re-render. If we select a facet value, we should re-render all results, even if their object is the same

Reply:
@olamothe if we have to fix this, then we could also reuse that value here instead

Reply:
> If we select a facet value, we should re-render all results, even if their object is the same

Ok, I see now, hmm. The thing that I don't like is that adding a property to every result feels like a work-around to a deeper issue. Every use-case is going to encounter the same problem, and will need to do something similar.

If there a was a utility function that could calculate an id based on the relevant parts of the request and response, then maybe other use-cases could use it. This id generator could be exposed on the interactive result controller as @olamothe suggested. Developers would see an official way to get the key in the reference docs, instead of concatenating two strings.